packages/cli/src/metadataGeneration/parameterGenerator.ts

```diff
@@ -305,6 +318,10 @@ export class ParameterGenerator {
     return undefined;
   }
 
+  private getParameterDeprecation(node: ts.ParameterDeclaration) {
+    return isExistJSDocTag(node, tag => tag.tagName.text === 'deprecated') || isDecorator(node, identifier => identifier.text === 'Deprecated');
```

Comment:
I'm a little torn here, the other (JSDoc) modifiers don't have corresponding Parameter Decorators.

Reply:
I don't feel too strongly, I can leave off the decorator logic here if you like. Since TS supports decorating parameters I figured why not give people the option. I do think it's good to keep the JSDoc annotation support here since people already use that in the wild and it's supported by IDEs/etc.

Reply:
I think we keep it, I can't see a case where it's causing confusion especially compared to the issue we'd inevitably get eventually when someone tries to do it using decorators. Intent is clear and it doesn't interfere with anything else.